tests/ui/transmute_ptr_to_ptr.stderr

```diff
@@ -1,3 +1,17 @@
+error: transmute from a type (`&T`) to itself
+  --> $DIR/transmute_ptr_to_ptr.rs:8:5
+   |
+LL |     std::mem::transmute::<&'a T, &'static T>(t)
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::useless-transmute` implied by `-D warnings`
```

Comment:
I think lifetime transmute were intentionally not linted before.
https://github.com/rust-lang/rust-clippy/blob/23549a8c362a403026432f65a6cb398cb10d44b7/tests/ui/transmute_ptr_to_ptr.rs#L47